### PR TITLE
fix: add creation of artifacts to spl_build

### LINKED ISSUE #133: Integrate creation of artifacts

### PR DESCRIPTION
**Pull request by JuReMq**

### CODE REVIEW

File: src/spl_core/test_utils/spl_build.py [hunk 1]
```diff
@@ -75,3 +103,66 @@ def execute(self, target: str, additional_args: Optional[List[str]] = None) -> i
...
+        """
+        artifacts_collection = ArtifactsCollection(artifacts=expected_artifacts, build_dir=self.build_dir)
+        json_content = {
+            "variant": self.variant,
+            "build_kit": self.build_kit,
+            "artifacts": [str(artifact.archive_path.as_posix()) for artifact in artifacts_collection.archive_artifacts],
+        }
+        json_path = self.build_dir / "artifacts.json"
+
+        # Delete the file if it already exists
```
Comment (cuinixam) [inline src/spl_core/test_utils/spl_build.py]: Simplify the code to only write the json content: `json_path.write_text(json.dumps(json_content, indent=4))`

File: tests/test_spl_build.py [hunk 2]
```diff
@@ -67,3 +71,33 @@ def test_execute_with_additional_args(mock_execute: MagicMock, spl_build: SplBui
 
     # Assertions
     mock_execute.assert_called_once_with(["build.bat", "-buildKit", "defaultKit", "-variants", "my_var", "-target", "all", "-reconfigure", "-j", "4"])
+
+
+def test_create_artifacts_archive(spl_build: SplBuild) -> None:
+    """
+    Test that additional arguments are passed to the command line executor correctly.
```
Comment (cuinixam) [inline tests/test_spl_build.py]: Does this description match the test case?

File: tests/test_spl_build.py [hunk 3]
```diff
@@ -67,3 +71,33 @@ def test_execute_with_additional_args(mock_execute: MagicMock, spl_build: SplBui
...
+        assert file_list == ["some_folder/blub.exe", f"out/{file_1.name}", file_2.name, outside_file.name]
+    assert archive_json.exists()
+    assert dict(json.loads(archive_json.read_text())) == {'variant': 'my_var',
+                                                          'build_kit': 'defaultKit',
+                                                          'artifacts': [
+                                                              "some_folder/blub.exe",
+                                                              f"out/{file_1.name}",
+                                                              file_2.name,
+                                                              outside_file.name]
+                                                          }
```
Comment (cuinixam) [inline tests/test_spl_build.py:131]: This test is not readable. I can not understand it. Try to split it in somehow in three parts: inputs, run, assertions

**Review by cuinixam — CHANGES_REQUESTED** [review]

File: tests/test_spl_build.py (hunk 3 above)
Comment (JuReMq) [reply tests/test_spl_build.py:131]: Updated tests

**Review by cuinixam — APPROVED** [review]

**Pull request merged**